bwscanner/attacher.py

```diff
@@ -162,3 +147,46 @@ def add_single_port_exit(port):
                               'ExitPolicy', 'accept 127.0.0.1:{}, reject *:*'.format(port))
     return port.addCallback(add_single_port_exit).addCallback(
         lambda ign: tor.routers[tor.protocol.get_info("fingerprint")])
+
+
+@defer.inlineCallbacks
+def connect_to_tor(launch_tor, circuit_build_timeout, circuit_idle_timeout, control_port=9051):
+    """
+    Launch or connect to a Tor instance
+
+    Configure Tor with the passed options and return a Deferred
+    """
+    # Options for spawned or running Tor to load the correct descriptors.
+    tor_options = {
+        'LearnCircuitBuildTimeout': 0,  # Disable adaptive circuit timeouts.
+        'CircuitBuildTimeout': circuit_build_timeout,
+        'CircuitIdleTimeout': circuit_idle_timeout,
+        'UseEntryGuards': 0,  # Disable UseEntryGuards to avoid PathBias warnings.
+        'UseMicroDescriptors': 0,
```

Comment:
Setting UseMicroDescriptors causes Tor to fetch new descriptors and Txtorcon then updates it's TorState object. There is a race where this state update can cause the tests to fail.